src/app/location-spoofing/api/activate-ppp/route.ts

```diff
@@ -0,0 +1,74 @@
+import { NextResponse } from 'next/server';
+import { getAndValidateFingerprintResult } from '../../../../server/checks';
+import { getRegionalDiscount } from '../../data/getDiscountByCountry';
+import { env } from '../../../../env';
+import { LOCATION_SPOOFING_COPY } from '../../copy';
+import { getIpLocation } from '../../../../shared/utils/locationUtils';
+
+export type ActivateRegionalPricingPayload = {
+  requestId: string;
+  sealedResult?: string;
+};
+
+export type ActivateRegionalPricingResponse =
+  | { severity: 'success'; message: string; data: { discount: number } }
+  | {
+      severity: 'warning' | 'error';
+      message: string;
+    };
+
+export async function POST(req: Request): Promise<NextResponse<ActivateRegionalPricingResponse>> {
+  const { requestId, sealedResult } = (await req.json()) as ActivateRegionalPricingPayload;
+
+  // Get the full Identification result from Fingerprint Server API and validate its authenticity
+  const fingerprintResult = await getAndValidateFingerprintResult({
+    requestId,
+    req,
+    sealedResult,
+    serverApiKey: env.SEALED_RESULTS_SERVER_API_KEY,
+  });
+  if (!fingerprintResult.okay) {
+    return NextResponse.json({ severity: 'error', message: fingerprintResult.error }, { status: 403 });
+  }
+
+  const country = getIpLocation(fingerprintResult.data)?.country;
+  const vpnDetection = fingerprintResult.data.products?.vpn?.data;
+
+  if (!country) {
+    return NextResponse.json(
+      {
+        severity: 'error',
+        message: 'Location could not be determined, please try a different browser or internet connection.',
+      },
+      { status: 403 },
+    );
+  }
+
+  if (vpnDetection?.result === true) {
+    let reason = '';
+    if (vpnDetection.methods?.publicVPN) {
+      reason = `Your IP address appears to be in ${country.name} but it's a known VPN IP address.`;
+    }
+    if (vpnDetection.methods?.timezoneMismatch && vpnDetection.originTimezone) {
+      reason = `Your IP address appears to be in ${country.name}, but your timezone is ${vpnDetection.originTimezone}.`;
```

Comment:
Maybe this could be worded a little differently since the time zone could still be in the same country. Example: I'm in Seattle using a VPN in Chicago and get this: "Your IP address appears to be in United States, but your timezone is America/Los_Angeles." I'm still in the US despite being in the wrong time zone.